corehq/apps/cloudcare/views.py

```diff
@@ -406,6 +410,76 @@ def get(self, request, *args, **kwargs):
         })
 
 
+@location_safe
+class LoginAsUsers(View):
+
+    http_method_names = ['get']
+    urlname = 'login_as_users'
+
+    @method_decorator(login_and_domain_required)
+    @method_decorator(require_can_edit_commcare_users)
+    @method_decorator(requires_privilege_for_commcare_user(privileges.CLOUDCARE))
+    def dispatch(self, *args, **kwargs):
+        return super(LoginAsUsers, self).dispatch(*args, **kwargs)
+
+    @property
+    @memoized
+    def can_access_all_locations(self):
+        return self.couch_user.has_permission(self.domain, 'access_all_locations')
+
+    def get(self, request, domain, **kwargs):
+        self.domain = domain
+        self.couch_user = request.couch_user
+
+        try:
+            limit = int(request.GET.get('limit', 10))
+        except ValueError:
+            limit = 10
+
+        # front end pages start at one
+        try:
+            page = int(request.GET.get('page', 1))
+        except ValueError:
+            page = 1
+        query = request.GET.get('query')
+
+        users_query = self._user_query(query, page - 1, limit)
+        total_records = users_query.count()
+        users_data = users_query.run()
+
+        return json_response({
+            'response': {
+                'itemList': map(lambda user: self._format_user(user), users_data.hits),
+                'total': users_data.total,
+                'page': page,
+                'query': query,
+                'total_records': total_records
+            },
+        })
+
+    def _user_query(self, search_string, page, limit):
+        return login_as_user_query(
```

Comment:
i'm just going to conditionally pass in an array of custom user data fields based on whether or not the ENIKSHAY toggle is enabled. if we ever need to search certain data props for other domains we can make another toggle